packages/cli/src/commands/pull.ts

```diff
@@ -77,32 +81,93 @@ export default async function main(client: Client) {
 
     if (link.status === 'not_linked') {
       // User aborted project linking questions
-      return 0;
+      return { linkResultCode: 0 };
     }
   }
 
   if (link.status === 'error') {
-    return link.exitCode;
+    return { linkResultCode: link.exitCode };
   }
 
-  const { project, org } = link;
+  return { link, linkResultCode: 0 };
+}
 
-  client.config.currentTeam = org.type === 'team' ? org.id : undefined;
+async function pullAllEnvFiles(
+  envFileRoot: string,
+  client: Client,
+  project: Project,
+  argv: any,
```

Comment:
The dynamic type `argv` can't be referenced in code. I could manually create a matching type, but that would be brittle. Is there any way to get that type into this function?

Reply:
Other than inlining the function definition into scope and using `typeof`, I don't think so unless you can abstract and pass a generic to share in both places

Reply:
I extracted the arg parsing into a separate function that can return that dynamic type, then used ` ReturnType<typeof processArgs>` to get a reference to it here. Seems to work!